common/src/main/java/com/wynntils/core/persisted/storage/StorageManager.java

```diff
@@ -130,8 +130,12 @@ private void writeToJson() {
         JsonObject storageJson = new JsonObject();
 
         storages.forEach((jsonName, storage) -> {
-            JsonElement jsonElem = Managers.Json.GSON.toJsonTree(storage.get(), storageTypes.get(storage));
-            storageJson.add(jsonName, jsonElem);
+            try {
+                JsonElement jsonElem = Managers.Json.GSON.toJsonTree(storage.get(), storageTypes.get(storage));
+                storageJson.add(jsonName, jsonElem);
+            } catch (Throwable t) {
+                WynntilsMod.error("Failed to save storage " + jsonName, t);
+            }
```

Comment:
Only used this for debugging, but I think it should be kept, to reduce the risk of "resetting" storage files.

Reply:
Yes, I have seen some people reporting their storage file getting reset so hopefully this will help prevent that